src/common/mmap_linux.c

```diff
@@ -65,8 +71,8 @@ util_map_hint_unused(void *minaddr, size_t len, size_t align)
 	ASSERT(align > 0);
 
 	FILE *fp;
-	if ((fp = os_fopen("/proc/self/maps", "r")) == NULL) {
-		ERR("!/proc/self/maps");
+	if ((fp = os_fopen(OS_MAPFILE, "r")) == NULL) {
+		ERR("!%s", OS_MAPFILE);
```

Comment:
I looked up where this will be used, as I see:
This return can return MAP_FAILED to `util_map_hint`, which in turn can return that to `util_map` in `src/common/mmap.c` where this can result in a failure:
```
     void *addr = util_map_hint(len, req_align);                         
     if (addr == MAP_FAILED) {                                           
             ERR("cannot find a contiguous region of given size");       
             return NULL;                                                
     }                                                                   
```
Shouldn't that part in util_map be modified, I mean making the hint thing optional, for the case of not having procfs on FreeBSD?